src/array.php

```diff
@@ -59,3 +59,24 @@ function to_array($value)
 
     return iterator_to_array($value);
 }
+
+/**
+ * Extract a specific key from each array
+ *
+ * NULL values are removed from results
+ *
+ * @param array $source
+ * @param string $key
+ *
+ * @return array
+ */
+function pluck(array $source, $key)
+{
+    $array = array_map(function ($array) use ($key) {
+        if (isset($array[$key])) {
+            return $array[$key];
+        }
+    }, $source);
+
+    return array_filter($array);
```

Comment:
I don't think this part is desired or wanted, really. It will nuke `0` values, for instance.